checker.c

```diff
@@ -1,21 +1,17 @@
 #include <stdio.h>
 #include <assert.h>
+#include "header.h"
+
+int count;
 
 int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+  count = isTempNotInRange(temperature);
+  count = isStateNotInRange(soc);
+  count = isChargeNotInRange(chargeRate);
+  return count;
 }
 
 int main() {
```

Comment:
These tests do not cover all the written conditions. Add more tests.
Hint: The number of tests can be minimized by avoiding duplicate checks in the code - see comment above.